cni/network/invoker_azure.go

```diff
@@ -65,7 +70,11 @@ func (invoker *AzureIPAMInvoker) Add(addConfig IPAMAddConfig) (IPAMAddResult, er
 	defer func() {
 		if err != nil {
 			if len(addResult.ipv4Result.IPs) > 0 {
-				if er := invoker.Delete(&addResult.ipv4Result.IPs[0].Address, addConfig.nwCfg, nil, addConfig.options); er != nil {
+				addresses := []*net.IPNet{}
+				for _, ip := range addResult.ipv4Result.IPs {
```

Comment:
Here we are appending from ipv4 result. And in the delete function we are deleting the ipv6 addresses as well. Correct me if am wrong but shouldn't we have the ipv6 addresses in the list too ?
OR am i missing something ? we can add a comment describing what the intent of this function is.

Reply:
Hi, I missed this comment, yes, we do need delete ipv6 addresses here as well